toolkit/tools/pkg/imagecustomizerlib/customizeboot.go

```diff
@@ -128,6 +178,31 @@ func setLinuxPath(inputGrubCfgContent string, linuxPath string) (outputGrubCfgCo
 	return outputGrubCfgContent, oldKernelPath, nil
 }
 
+// Overrides the path of the kernel binary of all the linux commands within a grub config file.
+func setLinuxPathAll(inputGrubCfgContent string, linuxPath string) (outputGrubCfgContent string, err error) {
+	quotedLinuxPath := grub.QuoteString(linuxPath)
+
+	lines, err := findLinuxLineAll(inputGrubCfgContent)
+	if err != nil {
+		return "", err
+	}
+
+	outputGrubCfgContent = inputGrubCfgContent
+	// loop from last to first so that the captured locations from
+	// findGrubCommandAll are not invalidated as reconstructing
+	// outputGrubCfgContent.
+	for i := len(lines); i > 0; i = i - 1 {
```

Comment:
Another way of doing this is to rebuild the string from scratch. See `updateKernelCommandLineArgsHelper` as an example.

Reply:
we could - I'm thinking we can just leave it this way.

Reply:
Fair